src/neuroglancer/util/geom.ts

```diff
@@ -148,3 +149,35 @@ export function translationRotationScaleZReflectionToMat4(
   out[2] = scale[2] * zReflection;
   return mat4.fromRotationTranslationScale(out, rotation, translation, <vec3>temp);
 }
+
+
+/**
+ * Transforms a z-index to/from Morton Codes (i.e. z-indices).
+ */
+const magicBits = [
+  new Uint64(0x49249249, 0x12492492), // AND step
+  new Uint64(0xc30c30c3, 0x30c30c30), // XOR steps
+  new Uint64(0x0f00f00f, 0xf00f00f0),
+  new Uint64(0xff0000ff, 0x00ff0000),
+  new Uint64(0x0000ffff, 0x00ff0000),
+  new Uint64(0x001fffff, 0x00000000),
+];
+
+export function compactMorton(input: Uint64) {
```

Comment:
Yeah, the Morton decoding likely isn't a performance bottleneck. I just grouped 'em in a global const out of habit, though they aren't exported so they shouldn't pollute the global namespace. That, and you can loop through 'em for more compact code :)